### DISCUSSION TITLE
Discussion of the practicality of components

### DISCUSSION THREAD

**Discussion by SiroSong** [saadeghi/daisyui#3255] [General]
This is a beautiful UI component library, and the types of components are also very rich. When looking at the example code, I found a problem, that is, the components blindly use CSS to complete the component functions, which adds some unnecessary component structures.
For example, Accordion components (and other components that require state maintenance), the state of these components is maintained through an input element, which is not originally needed, and in a real scenario, this results in the state in my code and the state on which the component performance depends are not exactly equivalent.I don't know if adding additional DOM structure and state will burden the maintainability of my project.
So I'm confused whether this library can really improve developer productivity.🙂

**Comment by saadeghi** [comment]
The inputs are not extra. They are used to show/hide elements.  
For example in accordion examples, you can choose which one is open by default by checking the radio input.  
So the same way you copy an example and you edit the text content, you can choose what to so with the inputs.  

If you have a better idea about how I can provide the code examples, I would be happy to do so.
